doc/how-to/vshard_quick.rst

```diff
@@ -1,160 +1,525 @@
+
+Step 3: Configuring bucket count
+********************************
+
+Specify the total number of buckets in a sharded cluster using the ``sharding.bucket_count`` option:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-after: login: storage
+    :end-at: bucket_count
+    :dedent:
+
+
+..  _vshard-quick-start-configuring-cluster-topology:
+
+Step 4: Defining the cluster topology
+*************************************
+
+Define the cluster's topology inside the :ref:`groups <configuration_reference_groups>` section.
+The cluster includes two groups:
+
+*   ``storages`` includes two replica sets. Each replica set contains two instances.
+*   ``routers`` includes one router instance.
+
+Here is a schematic view of the cluster's topology:
+
+.. code-block:: yaml
+
+    groups:
+      storages:
+        replicasets:
+          storage-a:
+            # ...
+          storage-b:
+            # ...
+      routers:
+        replicasets:
+          router-a:
+            # ...
+
+1.  To configure storages, add the following code inside the ``groups`` section:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+        :language: yaml
+        :start-at: storages:
+        :end-before: routers:
+        :dedent:
+
+    The main group-level options here are:
+
+    *   ``app``: The ``app.module`` option specifies that code specific to storages should be loaded from the ``storage`` module. This is explained below in the :ref:`vshard-quick-start-storage-code` section.
+    *   ``sharding``: The ``sharding.roles`` option specifies that all instances inside this group act as storages.
+        A rebalancer is selected automatically from two master instances.
+    *   ``replication``: The :ref:`replication.failover <configuration_reference_replication_failover>` option specifies that a leader in each replica set should be specified manually.
+    *   ``replicasets``: This section configures two replica sets that constitute cluster storages.
+
+
+2.  To configure a router, add the following code inside the ``groups`` section:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+        :language: yaml
+        :start-at: routers:
+        :end-at: 127.0.0.1:3300
+        :dedent:
+
+    The main group-level options here are:
+
+    *   ``app``: The ``app.module`` option specifies that code specific to a router should be loaded from the ``router`` module. This is explained below in the :ref:`vshard-quick-start-router-code` section.
+    *   ``sharding``: The ``sharding.roles`` option specifies that an instance inside this group acts as a router.
+    *   ``replicasets``: This section configures one replica set with one router instance.
+
+
+Resulting configuration
+***********************
+
+The resulting cluster configuration should look as follows:
```

Comment:
Maybe add the name `config.yaml` name in this sentence or the example itself for those readers who just scroll and quickly scan the page?

Reply:
Changed to:

```
The resulting ``config.yaml`` file should look as follows:
```